site/public/ts/digital/controllers/selectionpopup/OutputCountPopupModule.ts

```diff
@@ -51,11 +52,15 @@ export class OutputCountPopupModule extends SelectionPopupModule {
 
     public push(): void {
         const selections = this.circuitController.getSelections() as Encoder[];
-        const countAsNumber = this.count.valueAsNumber;
-
-        this.circuitController.addAction(new GroupAction(
-            selections.map(o => new OutputPortChangeAction(o, countAsNumber))
-        ).execute());
+        let countAsNumber = this.count.valueAsNumber;
+        if (countAsNumber % 1 === 0) // cannot be NaN
+        {
+            countAsNumber = Clamp(countAsNumber, parseInt(this.count.min), parseInt(this.count.max));
+            this.count.value = countAsNumber.toString();
```

Comment:
This is being repeated so many times, we should make a shared method somewhere
My recommendation:

Make a sub-class of `SelectionPopupModule` called `NumberInputPopupModule` that the 3 modules will extend instead and it will have a
`    protected count: HTMLInputElement;` instance variable and maybe you can reduce the repetition of code in `OutputCountPopupModule`, `InputCountPopupModule`, and `ClockFrequencyPopupModule`, and for sure have a `getInput()` method which returns the sanitized and clamped input